tests/test_init.py

```diff
@@ -53,3 +57,60 @@ def test_parse_bad_patchwork_url(self):
             "Can't parse patchwork url: '{}'".format(patchwork_url)
             in context.exception
         )
+
+    def test_ktree_init(self):
+        """Test ktree.__init__ without specifying a work directory"""
+        result = skt.ktree(GIT_URI)
+        self.assertIsInstance(result, skt.ktree)
+        self.assertEqual(result.uri, GIT_URI)
+
+    def test_ktree_init_workdir(self):
+        """Test ktree.__init__ with specifying a work directory"""
+        workdir = tempfile.mkdtemp()
+        result = skt.ktree(GIT_URI, wdir=workdir)
+        self.assertIsInstance(result, skt.ktree)
+        self.assertEqual(result.uri, GIT_URI)
+        self.assertEqual(result.wdir, workdir)
+
+    def test_ktree_create_workdir(self):
+        """Ensure create_workdir() creates a work dir if it does not exist"""
+        workdir = tempfile.mkdtemp()
+        os.rmdir(workdir)
+        result = skt.ktree(GIT_URI, wdir=workdir)
+        self.assertIsInstance(result, skt.ktree)
+        self.assertEqual(result.uri, GIT_URI)
+        self.assertEqual(result.wdir, workdir)
```

Comment:
Please do not test everything again and again in every test. If you have to test something, then have that as an explicit separate test. Make them independent and orthogonal. Otherwise you will have all these tests fail when one thing they all repeat fails, and that's not helpful.